Rules/AvoidUnusableParameter.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// AvoidUnusableParameter: Check if a parameter has a unusable name.
+    /// </summary>
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    public class AvoidUnusableParameter : IScriptRule
+    {
+        /// <summary>
+        /// AnalyzeScript: Check if a parameter has a unusable name.
+        /// </summary>
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            // Finds all functionAst
+            IEnumerable<Ast> functionAsts = ast.FindAll(testAst => testAst is FunctionDefinitionAst, true);            
+            var regex = new Regex(@"^\d+$");
+
+            foreach (FunctionDefinitionAst funcAst in functionAsts)
+            {
+                if (funcAst.Body != null && funcAst.Body.ParamBlock != null
+                    && funcAst.Body.ParamBlock.Attributes != null && funcAst.Body.ParamBlock.Parameters != null)
+                {
+                    foreach (var paramAst in funcAst.Body.ParamBlock.Parameters)
+                    // TODO: We need to iterate also EndBlock for statement parameters
+                    {
+                        bool unusableName = false;
+                        string paramName = paramAst.Name.VariablePath.UserPath;
+
+                        // check if name contains only digits
+
+                        if (regex.Match(paramName).Success) 
+                        {
+                            unusableName = true;
+                        }
+                        else if (paramName.StartsWith("|"))
+                        {
+                            unusableName = true;
+                        }
+
+                        // yield warning

```

Comment:
this comment is not useful and just stating the obvious

Reply:
Removed